packages/directory/tests/integration/components/dir-table-test.js

```diff
@@ -77,6 +77,105 @@ module('Integration | Component | dir table', function(hooks) {
     );
   });
 
+  test('table is sorted correctly', async function(assert) {
+    assert.expect(2);
+
+    set(this, 'items', []);
+    set(this, 'searchQuery', '');
+    set(this, 'sortBy', 'updatedOn');
+    set(this, 'sortDir', 'desc');
+
+    await render(hbs`{{dir-table
+      items=items
+      searchQuery=searchQuery
+      sortBy=sortBy
+      sortDir=sortDir
+    }}`);
+
+    let th = [...this.element.querySelectorAll('th')];
+
+    assert.deepEqual(
+      th.map(elm => elm.className.includes('is-sortable')),
+      [true, false, true, true],
+      'Only title, author and last update columns are sortable'
+    );
+
+    assert.deepEqual(
+      th.map(elm => {
+        let i = elm.querySelector('i');
+        return i ? i.className.includes('desc') : false;
+      }),
+      [false, false, false, true],
+      'Last update column is sorted'
+    );
+  });
+
+  test('onColumnClick action is called on column click', async function(assert) {
+    assert.expect(3);
+
+    set(this, 'items', []);
+    set(this, 'searchQuery', '');
+    set(this, 'sortBy', 'author');
+    set(this, 'onColumnClick', sort => {
+      assert.deepEqual(
+        sort,
+        { sortBy: 'author', sortDir: 'desc' },
+        'onColumnClick is called with desc when clicking an ascending-sorted column'
+      );
+    });
+
+    await render(hbs`{{dir-table
+      items=items
+      searchQuery=searchQuery
+      sortBy=sortBy
+      onColumnClick=onColumnClick
+    }}`);
+
+    run(() => {
+      this.element.querySelector('th:nth-child(3)').click();
+    });
+
+    set(this, 'onColumnClick', sort => {
+      assert.deepEqual(
+        sort,
+        { sortBy: 'updatedOn', sortDir: 'asc' },
+        'onColumnClick is called with asc when clicking an unsorted column'
+      );
+    });
+
+    await render(hbs`{{dir-table
+      items=items
+      searchQuery=searchQuery
+      sortBy=sortBy
+      onColumnClick=onColumnClick
+    }}`);
+
+    run(() => {
+      this.element.querySelector('th:nth-child(4)').click();
+    });
+
+    set(this, 'sortDir', 'desc');
+    set(this, 'onColumnClick', sort => {
+      assert.deepEqual(
+        sort,
+        { sortBy: 'author', sortDir: 'asc' },
+        'onColumnClick is called with asc when clicking a descending-sorted column'
```

Comment:
Should clicking a `desc` sorted column return the column to unsorted?

Reply:
Yeah I think it would be best to have it cycle between all 3 possible states

Reply:
@mattgraham1995 In reports we always have date in the sort which we default to, however here I'm not sure about the UX - should the table be sorted by title by default? what happens when a user clicks the author header for the 3rd time - the table is sorted by title again? not sure it's very intuitive

if you look at most 3rd part tables they don't have a 3rd state

Reply:
I suppose we do need to have some column sorted by default so 3 states aren't really necessary.

Reply:
@mattgraham1995 one thing I did add is `sortDescFirst` for columns like `updatedOn` since it makes more sense to sort desc on first click, can you take another look?